app/models/unit.rb

```diff
@@ -1,26 +1,21 @@
 class Unit < ActiveRecord::Base
 
   belongs_to :system
+  belongs_to :destination, class_name: "System", foreign_key: :destination_id
 
   validates :name, presence: true
 
-  def initialize
-    system.contents << self
-    @destination = system
-  end
-
-
-  def set_destination(destination)
-    if @location.links.include?(destination)
-      @destination = destination
+  def set_destination(target)
+    if system.links.include?(target)
```

Comment:
is links an association? if so, utilize SQL instead of in memory calculations here.

```ruby
if system.links.where(target: target).count > 0
```